src/gtk/webview_webkit2.cpp

```diff
@@ -325,49 +289,55 @@ wxgtk_webview_webkit_title_changed(GtkWidget* widget,
 }
 
 static void
-wxgtk_webview_webkit_resource_req(WebKitWebView *,
-                                  WebKitWebResource *,
-                                  WebKitURIRequest *request,
-                                  wxWebViewWebKit *webKitCtrl)
+wxgtk_webview_webkit_uri_scheme_request_cb(WebKitURISchemeRequest *request,
+                                           wxWebViewWebKit *webKitCtrl)
 {
-    wxString uri = webkit_uri_request_get_uri(request);
+    wxString scheme = webkit_uri_scheme_request_get_scheme(request);
 
     wxSharedPtr<wxWebViewHandler> handler;
     wxVector<wxSharedPtr<wxWebViewHandler> > handlers = webKitCtrl->GetHandlers();
 
-    //We are not vetoed so see if we match one of the additional handlers
     for(wxVector<wxSharedPtr<wxWebViewHandler> >::iterator it = handlers.begin();
         it != handlers.end(); ++it)
     {
-        if(uri.substr(0, (*it)->GetName().length()) == (*it)->GetName())
+        if(scheme == (*it)->GetName())
         {
             handler = (*it);
         }
     }
-    //If we found a handler we can then use it to load the file directly
-    //ourselves
+
     if(handler)
     {
-        //If it is requsting the page itself then return as we have already
-        //loaded it from the archive
-        if(webKitCtrl->m_vfsurl == uri)
-            return;
+        wxString uri = webkit_uri_scheme_request_get_uri(request);
 
         wxFSFile* file = handler->GetFile(uri);
         if(file)
         {
-            //We load the data into a data url to save it being written out again
-            size_t size = file->GetStream()->GetLength();
-            char *buffer = new char[size];
-            file->GetStream()->Read(buffer, size);
-            wxString data = wxBase64Encode(buffer, size);
-            delete[] buffer;
+            gint64 length = file->GetStream()->GetLength();
+            guint8 *data = g_new(guint8, length);
+            file->GetStream()->Read(data, length);
+            GInputStream *stream = g_memory_input_stream_new_from_data(data,
+                                                                       length,
+                                                                       g_free);
             wxString mime = file->GetMimeType();
-            wxString path = "data:" + mime + ";base64," + data;
-            //Then we can redirect the call
-            webkit_uri_request_set_uri(request, path.utf8_str());
+            webkit_uri_scheme_request_finish(request, stream, length, mime);
+        }
+        else
+        {
+            GError *error = g_error_new(WEBKIT_NETWORK_ERROR,
```

Comment:
It would be worth extending `wxGtkError` class from `include/wx/gtk/private/error.h` to have a non-default ctor and use it here to avoid manual calls to `g_error_free()`.

Reply:
Done.